usefulUtilities/tabStatus/appResources/appData/resources/js/tabStatus_ui.js

```diff
@@ -0,0 +1,606 @@
+            ${teamMembers}
+        </table>
+    `
+    teamContainer.appendChild(div);
+    input.focus();   
+}
+
+
+// show the status state
+function showStatus(){
+    let formContainer = document.getElementById('formContainer');
+    let teamContainer = document.getElementById('teamContainer');
+    let status = document.getElementById('status');
+    let input = document.getElementById('filter_members');
+    
+    formContainer.style.display = "block";
+    teamContainer.style.display = "none";
+    input.style.display = "none";
+
+    status.focus();
+}
+
+
+// show the table list state
+function showList(){
+    let input = document.getElementById('filter_members');
+    let formContainer = document.getElementById('formContainer');
+    let teamContainer = document.getElementById('teamContainer');
+    let status = document.getElementById('status');
+
+    formContainer.style.display = "none";
+    teamContainer.style.display = "block";
+    input.style.display = "block";
+
+    input.value = previousSearch;
+    status.value = "";
+    renderUI();
+}
+
+
+
+// #endregion
+// *************************************
+// END render
+// *************************************
+
+// *************************************
+// START constructors
+// *************************************
+// #region constructors
+
+
+// Worker Maker to help sort employees 
+function WorkerMaker(displayName, status, location, team) {
+    this.displayName = displayName;
+    this.status = status;
+    this.location = location;
+    this.team = team;
+}
+
+
+// #endregion
+// *************************************
+// END constructors
+// *************************************
+
+// *************************************
+// START filter-sort
+// *************************************
+// #region filter-sort
+
+
+// Check to see if we need to combine terms when we split the keyword up
+// This is only used if you want to exclude a term like Client and Engine. 
+// You have to use !(Client and engine)
+// this finds the first index of ( and the index with) and combines them together
+function handleKeywordArraySplit(keyword){
+    let keywordArray = keyword.split(" ");
+
+    let firstPar = hasInArray(keywordArray, "(");
+    let secondPar = hasInArray(keywordArray, ")");
+    if (firstPar > -1) {
+        let combined = keywordArray.slice(firstPar, secondPar+1).join(" ");
+
+        keywordArray = [
+            ...keywordArray.slice(0, firstPar),
+            combined,
+            ...keywordArray.slice(secondPar + 1)
+        ]
+    }
+
+    return keywordArray;
+}
+
+// O(1) check to see if the status keyword is being typed
+const statusObject = {
+    "s": true,
+    "st": true,
+    "sta": true,
+    "stat": true,
+    "statu": true,
+    "status": true
+}
+function statusCheckerForSearchStateSave(keyword){
+    let testKeyword = keyword.split(" ")[0]
+    // The keyword is greater than status so save and move on
+    if (testKeyword.length > 6) {

```

Comment:
can you please help me understand this logic?

Reply:
i believe you're trying to do some early optimization here. i'd argue that `"status".length` is better than `6`. i'd also argue that you want to test if `keyword.split(" ").length > 1`